src/servers/src/http/authorize.rs

```diff
@@ -12,118 +12,108 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::marker::PhantomData;
-
 use ::auth::UserProviderRef;
+use axum::extract::State;
 use axum::http::{self, Request, StatusCode};
-use axum::response::Response;
+use axum::middleware::Next;
+use axum::response::{IntoResponse, Response};
+use axum::Json;
 use base64::prelude::BASE64_STANDARD;
 use base64::Engine;
 use common_catalog::consts::DEFAULT_SCHEMA_NAME;
 use common_catalog::parse_catalog_and_schema_from_db_string;
 use common_error::ext::ErrorExt;
 use common_telemetry::warn;
-use futures::future::BoxFuture;
 use headers::Header;
-use http_body::Body;
 use secrecy::SecretString;
 use session::context::QueryContext;
 use snafu::{ensure, OptionExt, ResultExt};
-use tower_http::auth::AsyncAuthorizeRequest;
 
 use super::header::GreptimeDbName;
-use super::PUBLIC_APIS;
+use super::{JsonResponse, PUBLIC_APIS};
 use crate::error::{
     self, InvalidAuthorizationHeaderSnafu, InvalidParameterSnafu, InvisibleASCIISnafu,
     NotFoundInfluxAuthSnafu, Result, UnsupportedAuthSchemeSnafu, UrlDecodeSnafu,
 };
 use crate::http::HTTP_API_PREFIX;
 
-pub struct HttpAuth<RespBody> {
-    user_provider: Option<UserProviderRef>,
-    _ty: PhantomData<RespBody>,
+#[derive(Clone)]
+pub struct AuthState {
+    pub user_provider: Option<UserProviderRef>,
 }
 
-impl<RespBody> HttpAuth<RespBody> {
-    pub fn new(user_provider: Option<UserProviderRef>) -> Self {
-        Self {
-            user_provider,
-            _ty: PhantomData,
+pub async fn inner_auth<B>(
```

Comment:
```suggestion
pub(crate) async fn inner_auth<B>(
```

Reply:
`pub` here is for reference in tests like ` src/servers/tests/http/authorize.rs`.